scvi/model/_destvi.py

```diff
@@ -0,0 +1,282 @@
+
+        Parameters
+        ----------
+        st_adata
+            registed anndata object
+        sc_model
+            trained CondSCVI model
+        use_gpu
+            Use the GPU or not.
+        **model_kwargs
+            Keyword args for :class:`~scvi.model.DestVI`
+        """
+        decoder_state_dict = sc_model.module.decoder.state_dict()
+        px_decoder_state_dict = sc_model.module.px_decoder.state_dict()
+        px_r = sc_model.module.px_r.detach().cpu().numpy()
+
+        return cls(
+            st_adata,
+            sc_model.scvi_setup_dict_["categorical_mappings"]["_scvi_labels"][
+                "mapping"
+            ],
+            decoder_state_dict,
+            px_decoder_state_dict,
+            px_r,
+            sc_model.module.n_hidden,
+            sc_model.module.n_latent,
+            sc_model.module.n_layers,
+            **model_kwargs,
+        )
+
+    def get_proportions(
+        self, adata: AnnData = None, keep_noise: bool = False
+    ) -> np.ndarray:
+        """
+        Returns the estimated cell type proportion for the spatial data.
+
+        Shape is n_cells x n_labels OR n_cells x (n_labels + 1) if keep_noise.
+
+        Parameters
+        ----------
+        adata
+            the anndata to loop through (used only if amortization)
+        keep_noise
+            whether to account for the noise term as a standalone cell type in the proportion estimate.
+        """
+        if self.is_trained_ is False:
+            raise RuntimeError("Please train the model first.")
+
+        adata = self._validate_anndata(adata)
+
+        column_names = self.cell_type_mapping
+        if keep_noise:
+            column_names = np.append(column_names, "noise_term")
+
+        if self.module.amortization in ["both", "proportion"]:
+            stdl = self._make_data_loader(adata=adata)
+            prop_ = []
+            for tensors in stdl:
+                generative_inputs = self.module._get_generative_input(tensors, None)
+                prop_local = self.module.get_proportions(
+                    x=generative_inputs["x"], keep_noise=keep_noise
+                )
+                prop_ += [prop_local.cpu()]
+            data = np.array(torch.cat(prop_))
+        else:
+            logger.info(
+                "No amortization for proportions, ignoring adata and returning results for the full data"
+            )
+            data = self.module.get_proportions(keep_noise=keep_noise)
+
+        return pd.DataFrame(
+            data=data,
+            columns=column_names,
+            index=self.adata.obs.index,
+        )
+
+    def get_gamma(self, adata=None) -> np.ndarray:
+        """
+        Returns the estimated cell-type specific latent space for the spatial data.
+
+        Shape is n_cells x n_latent x n_labels.
+        """
+        if self.is_trained_ is False:
+            raise RuntimeError("Please train the model first.")
```

Comment:
flag

Reply:
see other comment, this should be a user warning. Honestly it might even make sense to write a global util, but for now just a warning and we can correct later.